src/main/java/org/folio/service/job/JobExecutionServiceImpl.java

```diff
@@ -54,16 +58,31 @@
 
   @Override
   public Future<JobExecution> getById(final String jobExecutionId, final String tenantId) {
-    return jobExecutionDao.getById(jobExecutionId, tenantId)
-      .compose(optionalJobExecution -> {
-        if (optionalJobExecution.isPresent()) {
-          return succeededFuture(optionalJobExecution.get());
+    Promise<JobExecution> jobExecutionPromise = Promise.promise();
+    jobExecutionDao.getById(jobExecutionId, tenantId)
+      .onComplete(optionalJobExecution -> {
+        if (optionalJobExecution.succeeded() && optionalJobExecution.result().isPresent()) {
+          JobExecution jobExecution = optionalJobExecution.result().get();
```

Comment:
we need to do this for get as well, may be get all the job profiles and just populate the job profile name for the jobs

Reply:
we could add cache later for the job profiles, so that DB is not called multiple times to get the names

Reply:
It sounds not really good if we are thinking about performance...why we need to do it for get endpoint? If the jobExecution is new, no jobProfileName there, if it is used for export - jobProfileName will be present in job execution. Maybe we can populate "default" jobProfileName for those jobExecutions, that don`t have this field and when this jobExecution will be used for export, the valid jobProfileName will be populated for it. What do you think?